### ISSUE TITLE
Issues with stratified_subsampling()

### ISSUE THREAD

**Issue by imenelydiaker** [embeddings-benchmark/mteb#519]
I ran into the following error when using our subsampling funciton on `GreekLegalCodeClassification` task:
```bash
... in train_test_split 
raise ValueError(
ValueError: The least populated class in label column has only 1 member, which is too few. The minimum number of groups for any class cannot be less than 2.
```

@isaac-chung it seems that `train_test_split` doesn't handle stratification when there is only one sample per class. Any idea on how to solve this?

**Comment by imenelydiaker** [comment]
I personally find it weird to have classes with only sample, maybe we shouldn't handle them? We can filter the dataset and remove rows with only 1 sample, wdyt? 
The shuffle will just don't consider the class imbalance and I'm not sure it's good to use it in a function we named `stratified_subsampling()` 🤔

**Comment by isaac-chung** [comment]
Good point. It wouldn't be true to the name anymore. I feel that we can put a note in this method to say that we will remove rows with only 1 sample. Use with caution.

**Comment by KennethEnevoldsen** [comment]
I will close this as not planned, but do feel free to re-open
